models/views/scratch/snowplow_unified_screen_summary_metrics.sql

```diff
@@ -0,0 +1,63 @@
+{#
+Copyright (c) 2024-present Snowplow Analytics Ltd. All rights reserved.
+This program is licensed to you under the Snowplow Personal and Academic License Version 1.0,
+and you may not use this file except in compliance with the Snowplow Personal and Academic License Version 1.0.
+You may obtain a copy of the Snowplow Personal and Academic License Version 1.0 at https://docs.snowplow.io/personal-and-academic-license-1.0/
+#}
+
+{{
+  config(
+    sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt'))
+  )
+}}
+
+with prep as (
+  select
+    ev.view_id
+    , ev.session_identifier
+
+    , round(max(ev.screen_summary__foreground_sec) * 100) / 100.0 as foreground_sec
+    , round(max(ev.screen_summary__background_sec) * 100) / 100.0 as background_sec
```

Comment:
Why this?

Reply:
I wasn't sure about this, but thought we should do some rounding because the value from trackers might come with too much precision. I went for rounding to 2 decimal points in the views table and to integers in the sessions and users table (see the `sessions_screen_summary_metrics` table).

Before all these values were integers because they were calculated based on the page pings which are in seconds. Happy to change to whatever you think makes sense, don't really know how we tend to do rounding in the package.

Reply:
It was more that round usually takes another argument for how many decimal places to round to :D

Reply:
*mind blown*

I had to cast the number to numeric – didn't work with float. I am quite confused as what that actually means though and whether I should be using type_float or type_numeric elsewhere...

```
round(cast(max(ev.screen_summary__foreground_sec) as {{ type_numeric() }}), 2)
```

Reply:
Gut feeling - you can round floats by it doesn't change their actual precision (unlike numerics), and you then end up dealing with float accuracy errors like 1.200000000000000001. I don't know if that's true but think I've had similar problems in the past